apps/prairielearn/elements/pl-xss-safe/pl-xss-safe.mustache

```diff
@@ -1,17 +1,15 @@
 <script>
-  $(function() {
+  $(async function() {
     const element = $('#xss-{{uuid}}');
     const language = element.attr("data-language");
     let code = element.attr("data-code");
     // Uses the same JS library as the pl-file-editor for markdown
     // conversion and xss filter for compatibility and to avoid
     // confusion
     if (language == "markdown") {
-      let renderer = new showdown.Converter({
-        'literalMidWordUnderscores':true,
-        'literalMidWordAsterisks':true,
-      });
-      code = renderer.makeHtml(code);
+      const marked = (await import('marked')).marked;
+      (await import('@prairielearn/marked-mathjax')).addMathjaxExtension(marked, MathJax);
+      code = await marked.parse(code);
```

Comment:
Does `marked.parse` return a promise or a value directly? The presence of `await` is inconsistent throughout this PR.

Reply:
Good catch! `marked.parse` [can return a promise if `marked` is initialized with `async: true`](https://marked.js.org/using_pro#async), but this isn't done in this context, so I'll remove it where it's not needed.